### PR TITLE
Added options to access private repos and github enterprise

### PR DESCRIPTION
**Pull request by MissingRoberto**
Hi, I found your extension pretty handy, but it only works for public repositories hosted on github.com.

* Added Popup for configuring parameters
* Popup style based on bootstrap
* Support for private repositories on github.com
* Support for Github Enterprise instances
* Support for Issue View

### CODE REVIEW

**Comment by marcbachmann** [comment]
Thanks for the pr. I've just done an iteration on top of your commit.
Here's the branch: https://github.com/marcbachmann/github-issue-templates/tree/pr/1
Wanna test it?

> Support for private registry on github.com

What do you mean by that? I thought private repos are supported because chrome just attaches the cookie of a user when doing the request to retrieve the template file.

**Comment by MissingRoberto** [comment]
Sorry, I meant private repositories, but you might be right about the cookie. I had just supposed that credentials would be required for that, as it's for Github Enterprise. 

Much better implementation with the domains. I actually thought about that, but I didn't have the time for it.

The options looks much better now with ratchet :) 

Nevertheless, it didn't work for me unfortunately: 
* You can't configure the alternative path to the templates
* I wasn't able to get the value of the domain

```
https://raw.undefined/Bluemix/cloudfoundry/master/.github/ISSUE_TEMPLATES.md net::ERR_NAME_NOT_RESOLVED

```

**Comment by marcbachmann** [comment]
I just pushed a working version onto the same branch.

**Comment by marcbachmann** [comment]
FYI. I don't like the user & password combination.
It also doesn't work properly if somebody is using 2FA.
I'd remove it and maybe support the access token creation using https://www.npmjs.com/package/github-create-token

File: extension/dist/manifest.json [hunk 1]
```diff
@@ -3,12 +3,18 @@
...
+    "default_popup": "options.html"
+  },
   "manifest_version": 2,
   "minimum_chrome_version": "58",
   "permissions": [
     "https://github.com/",
+    "https://github.ibm.com/",
+    "https://*.github.ibm.com/",
     "https://raw.githubusercontent.com/",
+    "https://*.github.com/",
```
Comment (marcbachmann) [inline extension/dist/manifest.json]: Is that also a url used by github for enterprise?

Comment (MissingRoberto) [reply extension/dist/manifest.json]: Not really, I was seeing tons of errors when trying to download things from `assets.github.com` maybe not related to this extension but it looked so to me.

Comment (MissingRoberto) [reply extension/dist/manifest.json]: Anyway, that didn't solve the issue.  Do you see also many errors to get assets and fonts in your browser?

Comment (marcbachmann) [reply extension/dist/manifest.json]: no, I don't have that.

Comment (marcbachmann) [reply extension/dist/manifest.json]: what error do you have there? 404's?

**Comment by MissingRoberto** [comment]
I have also added support for the dropdown in the `Issue View`. See d4f4cbd2dee15141fb33a0a4ab767c141a245779

**Comment by MissingRoberto** [comment]
Regarding the authentication, I also agree with that, I just didn't have the time to look at it.

**Comment by MissingRoberto** [comment]
[image: screenshot.png][image: screenshot.png]

**Comment by marcbachmann** [comment]
I can't see your screenshot.

**Comment by marcbachmann** [comment]
I just pushed a new version to the branch that dynamically requests permission.
The disadvantage of it is that you first have to request permission and then add the domain again because requesting the permission triggers a blur on the popup.

**Comment by marcbachmann** [comment]
Button support is also in there. Thanks for that.

**Comment by MissingRoberto** [comment]
Amazing now I am using this new branch instead of mine :)

About the requiring permissions... can't we try to be smarter? 

E.g: if this domain hasn't been added to our list of domains or it's not in the list of permission, then ask for permission when the domain matches a pattern `.*github*` or something like that. 

I didn't check how you request permission, just thinking of something.

These are the errors regarding the fonts:

screenshot


On the other hand, I was thinking about adding an additional template copying the content of the current one in the `issue view`, or an additional button `Copy Issue`. What do you think?

**Comment by marcbachmann** [comment]
> E.g: if this domain hasn't been added to our list of domains or it's not in the list of permission, then ask for permission when the domain matches a pattern .*github* or something like that.

Chrome extensions don't allow dynamic top level domains in a permission definition, so that won't work. I tried that :)

To prevent the blur when adding the domain, we could use a custom window instead of the popup.
But that way we would have to fiddle with the positioning. I also thought about adding a button next to the input field. But then we also have to save the current form in localstorage to resume to it once we reopen (which we also can't do I guess) the popover. 

> These are the errors regarding the fonts:

We actually use a woff font, but it's just one and not referenced in a frameworks file.
So the issue is probably somewhere on your side.

> On the other hand, I was thinking about adding an additional template copying the content of the current one in the issue view, or an additional button Copy Issue. What do you think?

Good idea. We can do that. I'd add a `Copy issue` link in the dropdown. Another button modifies the github ui too much. There are already enough buttons in there.

**Comment by MissingRoberto** [comment]
I saw another issue. 

In `View Issue` pages, the link to create new work items is wrong. It contains twice issues in the url

`https://github.com/org/repo/issues/issues/new?...`

**Comment by marcbachmann** [comment]
Thanks. I've used relative links before, so it didn't work for all urls.
I've switched to an absolute now.

**Comment by marcbachmann** [comment]
Merged & released in #2

**Pull request closed without merging**